### ISSUE TITLE
V11 - unexpected keyword argument 'groups'

### ISSUE THREAD

**Issue by didier1969** [it-projects-llc/odoo-saas-tools#711]
Hello,
I followed these instructions `https://odoo-saas-tools.readthedocs.io/en/latest/setup/install.html` as close as possible for Odoo 11 CE.

I am getting the error below at point 4 - Configure Main Database: * open Settings/SaaS Portal Settings, set Base SaaS domain, e.g. odoo.local, click Apply (do it even if you didn't make changes).

I wonder if this happens because V11 is still work in progress, and if I should better try V10 instead. Thanks for your help.
Didier

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 928, in call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 916, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/res/res_config.py", line 544, in execute
    self.set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/website/models/res_config_settings.py", line 84, in set_values
    super(ResConfigSettings, self).set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/auth_signup/models/res_config_settings.py", line 36, in set_values
    super(ResConfigSettings, self).set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/res_config_settings.py", line 44, in set_values
    super(ResConfigSettings, self).set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/base_setup/models/res_config_settings.py", line 57, in set_values
    self.env['ir.config_parameter'].sudo().set_param("base_setup.default_external_email_server", self.default_external_email_server)
  File "/home/dstadel/eclipse-workspace/odoo-custom-addons/saas_portal/models/ir_config_parameter.py", line 19, in set_param
    return super(IrConfigParameter, self).set_param(key, value, groups=groups)
TypeError: set_param() got an unexpected keyword argument 'groups'


Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 928, in call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 916, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/res/res_config.py", line 544, in execute
    self.set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/website/models/res_config_settings.py", line 84, in set_values
    super(ResConfigSettings, self).set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/auth_signup/models/res_config_settings.py", line 36, in set_values
    super(ResConfigSettings, self).set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/mail/models/res_config_settings.py", line 44, in set_values
    super(ResConfigSettings, self).set_values()
  File "/usr/lib/python3/dist-packages/odoo/addons/base_setup/models/res_config_settings.py", line 57, in set_values
    self.env['ir.config_parameter'].sudo().set_param("base_setup.default_external_email_server", self.default_external_email_server)
  File "/home/dstadel/eclipse-workspace/odoo-custom-addons/saas_portal/models/ir_config_parameter.py", line 19, in set_param
    return super(IrConfigParameter, self).set_param(key, value, groups=groups)
TypeError: set_param() got an unexpected keyword argument 'groups'
```

**Comment by gridworkz** [comment]
Hi:
Did you ever get this resolved? I'd like to hep here if possible.
Cheers, Dave

**Comment by didier1969** [comment]
Hello Dave,
No, I did not solve it yet and I switched to Odoo-Saas-Tool for Odoo V10 for now. This version 10 is working great so far. At your disposal if you need any additional information on V11.

**Comment by mpdevilleres** [comment]
You can easily fix this by removing the groups parameter.
this is due to the update in the base ir_config in v11.
Here's my ir_config_parameter.py.

    from odoo import models, api
    
    BASE_SAAS_DOMAIN = 'saas_portal.base_saas_domain'

    class IrConfigParameter(models.Model):
        _inherit = 'ir.config_parameter'
  
        @api.model
        def set_param(self, key, value):
            if key == 'web.base.url' and not self.get_param(BASE_SAAS_DOMAIN):
                domain_only = value
                if '/' in domain_only:
                    # get rid of "http(s)://" at the beggining and "/" and the end if any
                    domain_only = domain_only.split('/')[2]
               self.set_param(BASE_SAAS_DOMAIN, domain_only)
           return super(IrConfigParameter, self).set_param(key, value)

I hope it solves your issue.

**Comment by didier1969** [comment]
I updated the ir_config_parameter.py file from saas_portal module as per your instructions, and after deleting the databases and recreating them, I was able to have odoo.local talk to s1.odoo.local.
I am now experiencing another error message, i.e. in odoo.local, under Plans, when I click "Log in to Template DB". I also get the same error message when trying to create a new Client DB.

Many thanks in advance for your help !
Didier

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 944, in __get__
    value = record.env.cache.get(record, self)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 960, in get
    value = self._data[field][record.id][key]
KeyError: <odoo.api.Environment object at 0x7f7b65b54be0>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 924, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 916, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 687, in call_kw
    return call_kw_model(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 672, in call_kw_model
    result = method(recs, *args, **kwargs)
  File "/home/dstadel/eclipse-workspace/odoo-custom-addons/saas_portal/wizard/config_wizard.py", line 251, in default_get
    res['edit_database_url'] = active_record._request_url('/saas_server/edit_database')
  File "/home/dstadel/eclipse-workspace/odoo-custom-addons/saas_portal/models/saas_portal.py", line 516, in _request_url
    'host': r.host,
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 948, in __get__
    self.determine_value(record)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1059, in determine_value
    self.compute_value(recs)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1015, in compute_value
    self._compute_value(records)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1006, in _compute_value
    getattr(records, self.compute)()
  File "/home/dstadel/eclipse-workspace/odoo-custom-addons/saas_portal/models/saas_portal.py", line 450, in _compute_host
    base_saas_domain_1 = '.'.join(base_saas_domain.rsplit('.', 2)[-2:])
AttributeError: 'bool' object has no attribute 'rsplit'
```

**Comment by gridworkz** [comment]
Hi Didier:
Just a shot in the dark, the variable 'base_saas_domain' is false at this point of the code. I don't know why I'll keep poking around. Cheers, Dave

**Comment by gridworkz** [comment]
Hi Guys:
Is anyone up for making Saas tools work for Odoo 11? I've got some money and lot's of time to help make it happen. 
Cheers, Dave

**Comment by yelizariev** [comment]
@gridworkz and others,
you can use services like https://www.bountysource.com/ and https://gitcoin.co/ to sponsor updates.
Thank you

**Comment by gridworkz** [comment]
Hi:
Would it be easier if I just shoot you guys some cash to do it? Since you know the internals the best...
Cheers,
Dave

**Comment by ilmir-k** [comment]
Hi @gridworkz

Also you can use our paypal account paypal@it-projects.info if it is more convenient payment method for you. Thanks

**Comment by gridworkz** [comment]
Hi Guys:

I've got Saas Odo Toos running on v11. I think what we can do is I can give
you what I have and you can officially bless it and do the necessary things
to make it part of your repos (I didn't have enough time to learn the
process of contributing). I just cloned it and changed the appropriate
modules to make it work. (a base set of modules to make it work - I didn't
modify them all, some just worked but they were required deps).

access_limit_records_number
auth_oauth_check_client_id
auth_oauth_ip
product_price_factor
saas_base
saas_portal
saas_portal_demo
saas_portal_portal
saas_portal_sale
saas_portal_sale_online
saas_portal_signup
saas_portal_start
saas_portal_tagging
saas_portal_templates
saas_utils
website_sale_require_login
website_sale_stock_status
website_sale_suggest_create_account
website_seo_url
website_seo_url_product

https://drive.google.com/drive/folders/1GwBnU3ZrO7_r4QZombQhSIKjKi_yIwGd?usp=sharing

I had one warning in the logs regarding adding the base_saas_domain and
similar fields on that screen (it actually saves OK to the database it just
doesn't retain the value in the field after refreshing the page).

2018-01-09 02:23:32,004 5375 WARNING odoo.local
odoo.addons.base.res.res_config: Methods that start with `get_default_` are
deprecated. Override `get_values` instea
d(Method get_default_base_saas_domain)
2018-01-09 02:23:32,005 5375 WARNING odoo.local
odoo.addons.base.res.res_config: Methods that start with `get_default_` are
deprecated. Override `get_values` instea
d(Method get_default_expiration_notify_in_advance)
2018-01-09 02:23:32,006 5375 WARNING odoo.local
odoo.addons.base.res.res_config: Methods that start with `get_default_` are
deprecated. Override `get_values` instea
d(Method get_default_page_for_maximumdb)
2018-01-09 02:23:32,006 5375 WARNING odoo.local
odoo.addons.base.res.res_config: Methods that start with `get_default_` are
deprecated. Override `get_values` instea
d(Method get_default_page_for_maximumtrialdb)
2018-01-09 02:23:32,007 5375 WARNING odoo.local
odoo.addons.base.res.res_config: Methods that start with `get_default_` are
deprecated. Override `get_values` instea
d(Method get_default_page_for_nonfree_subdomains)

I don't know the code well enough to override those methods.

Hopefully this has sped up the process and helped you guys out. So the gig
would be to get this up to your standards - if you could quote on that it
would be appreciated.

Cheers,
Dave

**Comment by gridworkz** [comment]
Sorry, cat on my keyboard .... 'Saas Odoo Tools'....you probably already
guessed that :-)

**Comment by ilmir-k** [comment]
@gridworkz Thanks for your efforts, we will check it

cc @iledarn

**Comment by gridworkz** [comment]
Hi Guys:

Cool, thanks. One other thing I forgot to mention if it's relevant. I did a
git clone of the main Odoo v11 branch 3 days ago and saas_server worked out
of the box....

Cheers,
Dave

**Comment by 360org** [comment]
Hi @gridworkz have you found solution for SaaS Portal Portal it require Website Portal but only available for v10

**Comment by gridworkz** [comment]
Hi:

Yes, web_portal is now just portal in v11. It's pretty much a port of
web_portal. I find most things just work if you rename web_portal to portal
where it's referenced in the code. Some places you may need to fiddle a bit
to get it to work. I know that's not the official way of doing things but
it works for me while I'm shooting bugs.

Cheers,
Dave

**Comment by gridworkz** [comment]
Latest batch of files have been uploaded:
https://drive.google.com/drive/folders/1GwBnU3ZrO7_r4QZombQhSIKjKi_yIwGd?usp=sharing

**Comment by njeudy** [comment]
@didier1969 this error is corrected in #746 

I will try to clean code and adapt correctly to V11 .. (I use it already in V11)

**Comment by gridworkz** [comment]
Hi Nicolas:

That's great, much appreciated. Yes it seems OK on v11 at my end as well. I
should have done it through pull requests but it was a bit of a rush
situation at my end at the time.

Cheers
Dave
